### ISSUE TITLE
[router] Validate invalid combinations of routerrpc settings

### ISSUE THREAD

**Issue by guggero** [lightningnetwork/lnd#6553]
### Background

Originally reported here: https://github.com/lightninglabs/loop/issues/491

It looks like certain combinations of `routerrpc.*` settings can cause the path finding algorithm to search endlessly, using up a lot (and eventually all) memory available.

The settings in question were:

```
routerrpc.minrtprob=0.4
routerrpc.apriorihopprob=0.5
routerrpc.aprioriweight=0
```

My personal interpretation is that the combination of the min route probability of 40% with the a priori weight of 0 caused no routes to be found.

**Comment by Roasbeef** [comment]
> routerrpc.aprioriweight=0

Yeah this would mean that all hops w/o prior observations have a 0% chance of succeeding essentially.

**Comment by litbot-9000** [comment]
Closing as duplicate of #6910 which tracks removing the pathfinding probability floor. Unique details from this issue have been summarized there.
